src/Disks/ObjectStorages/LocalObjectStorage.cpp

```diff
@@ -104,13 +104,6 @@ std::unique_ptr<WriteBufferFromFileBase> LocalObjectStorage::writeObject( /// NO
     return std::make_unique<WriteBufferFromFile>(path, buf_size, flags);
 }
 
-void LocalObjectStorage::listPrefix(const std::string & path, RelativePathsWithSize & children) const
```

Comment:
Why did you remove that?

Reply:
It hadn't been compatible with the only user of this API - restore for `send_metadata`, for details see 333ac44c1483548e8e26daf050fcb98bf1cb478f

Reply:
Can you explain a bit more why it had been incompatible? I mean for `LocalObjectStorage` the implementation of those functions seems like it must be quite simple and standard.

Reply:
>Can you explain a bit more why it had been incompatible?

```
void LocalObjectStorage::listPrefix(const std::string & path, RelativePathsWithSize & children) const
{
    fs::directory_iterator end_it;
    for (auto it = fs::directory_iterator(path); it != end_it; ++it)
        children.emplace_back(it->path().filename(), it->file_size());
}
```

As you can see it does not uses `directory_recursive_iterator` here, and proper `listPrefix` should:
a) be recursive
b) return only files

Both of this wasn't satisfied.
Plus they are not used in the code anyway, since you cannot store metadata (`send_metadata`) in the local disk, and here is the piece of code that creates them:

```
DiskObjectStoragePtr DiskLocal::createDiskObjectStorage()
{
    auto object_storage = std::make_shared<LocalObjectStorage>();
    auto metadata_storage = std::make_shared<FakeMetadataStorageFromDisk>(
        /* metadata_storage */std::static_pointer_cast<DiskLocal>(shared_from_this()),
        object_storage,
        /* object_storage_root_path */getPath());

    return std::make_shared<DiskObjectStorage>(
        getName(),
        disk_path,
        "Local",
        metadata_storage,
        object_storage,
        false, // <-- here, this is send_metadata=false
        /* threadpool_size */16
    );
}
```

> I mean for LocalObjectStorage the implementation of those functions seems like it must be quite simple and standard.

Yes, it can, but I prefer to remove them to avoid ambiguity, anyway this code cannot be tested now.

Reply:
Yeah, I got it, so it was not incompatible, it was rather wrong.

I'd implement those functions for LocalObjectStorage anyway even if they probably won't be used right now. Just in the sake of consistency and kind of an alternative way to document those too functions better (because operations with filesystem can be considered as basic and simplest). Though I don't insist.

Reply:
My point is that implementing them w/o tests and w/o actual usage in the code is useless and can be even worse, since they can not work correctly and someone may rely on this.
Anyway I think that even if this should be done, this can be done separately.